src/Scope.ts

```diff
@@ -933,6 +933,63 @@ export class Scope {
                             range: container.callable.nameRange,
                             file: container.callable.file
                         });
+
+                        // Raise diagnostic error if the own callable and the shadowed callable
+                        // are of different types, e.g: sub vs. function.
+                        if (container.callable.isSub !== shadowedCallable.callable.isSub) {
+                            this.diagnostics.push({
+                                ...DiagnosticMessages.mismatchAncestorCallableType(
+                                    container.callable.name,
+                                    shadowedCallable.callable.isSub ? 'sub' : 'function'
+                                ),
+                                range: container.callable.nameRange,
+                                file: container.callable.file
+                            });
+                        }
+
+                        // Raise a diagnostic error if the own callable return type is different
+                        // than the shadowed callable.
+                        if (container.callable.type.returnType.toTypeString() !== shadowedCallable.callable.type.returnType.toTypeString()) {
```

Comment:
Should it be allowed to have a "more specific type" for the return type? Eg., use `isAssignableTo` ?

for example, shadowedCallable returns `dynamic`, but container returns `integer` - this is not technically an error, because container's return type (`integer`) can be assigned to a `dynamic`.

FWIW, Typescript uses assignability for evaluating if return types are valid on overridden functions